src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.TryGet.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.MemoryTests;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Memory.Tests
+{
+    public class ReadOnlySequenceTryGetTests
+    {
+        [Fact]
+        public void Ctor_Array_Offset()
```

Comment:
Can you add a test for empty array (i.e. `ReadOnlySequence<byte>(new byte[0])`)?

I don't see how this PR resolves the issue that @davidfowl mentioned here - https://github.com/dotnet/corefx/issues/27224

Reply:
It doesn't that's why I said that it's not a but in the issue you linked. This PR adds a way that should be use to get segment/index/array/memory which is TryGet* methods.

Reply:
I don't understand why it isn't a bug. Even though they are marked EditorBrowsableState.Never, these are public methods on a public type that anyone can call.

More importantly, ToString() is not hidden, and hence, if someone calls ToString() or looks at the state of SequencePosition in the debugger, the index value will not make any sense.

```C#
var ownedMemory = new CustomMemoryForTest<byte>(new byte[] { 1, 2, 3, 4, 5 }, 0, 5);
var buffer = new ReadOnlySequence<byte>(ownedMemory, 2, 3);
Console.WriteLine(buffer.Start.ToString());
// Actual: System.MemoryTests.CustomMemoryForTest`1[System.Byte][-2147483646]
// Expected: System.MemoryTests.CustomMemoryForTest`1[System.Byte][2]
```

Reply:
Design of `SequencePosition` is such that if you couldn't read it's properties outside `ReadOnlySequence`. You are right that having `ToString` makes less sense in this case so it should be removed.

We should look into adding `DebuggerDisplay` for `ReadOnlySequence` that would decode and display segments and indexes.

Reply:
I removed to string and renamed fields not to have index or segment in names.

Reply:
Filed https://github.com/dotnet/corefx/issues/27434